### ISSUE TITLE
Evaluate alternatives to storing executed notebooks in git

### ISSUE THREAD

**Issue by astrojuanlu** [poliastro/poliastro#727]
Doing widget state cleanup today:

```
$ git diff --stat
 docs/source/examples/Analyzing the Parker Solar Probe flybys.ipynb                  |  4295 ------------------------
 docs/source/examples/Catch that asteroid!.ipynb                                     |  2655 ---------------
 docs/source/examples/Exploring the New Horizons launch.ipynb                        |  1446 --------
 docs/source/examples/Going to Jupiter with Python using Jupyter and poliastro.ipynb | 11647 ----------------------------------------------------------------
 docs/source/examples/Using NEOS package.ipynb                                       |  6689 +------------------------------------
 docs/source/examples/Visualizing the SpaceX Tesla Roadster trip to Mars.ipynb       |   141 +-
 6 files changed, 106 insertions(+), 26767 deletions(-)
```

This was just the last straw, but storing executed notebooks takes a lot of space and makes code review more difficult. We should evaluate whether things like [jupytext](https://towardsdatascience.com/jupytext-1-0-highlights-a49dca9baa7b) are better suited for this.

**Comment by GorgiAstro** [comment]
At my work we just clear all notebook outputs before committing to git, but it does not solve everything because of Jupyter metadata: Python kernel version, table of content collapsed or not, etc. 

Therefore jupytext seems like an excellent solution for making version control easier. I tried it out, it is easy to use and just works :+1: 

However, even with jupytext the cell description contains the table of contents collapsed state of the cell: `+ {"toc-hr-collapsed": true, "cell_type": "markdown"} `
This is something to watch out for people like me who use the table of contents extension

**Comment by jorgepiloto** [comment]
When implementing #828 and after #976 it is clear that some notebooks do not seem to be updated even if a `make clean && make html` is executed. In ddition to that, reviewing modifications is difficult when it comes to `*.ipynb` as @astrojuanlu claimed.

I did a small ship locally from `*.ipynb` to `*.py` using Jupytext and found these advantages apart from previously ones:

* Notebooks are forced to run always, no matter if they do not raise API changes related errors.
* The recently implemented `nbsphinx-gallery` works without problem together with 3D plotting utilities.
* Syntax is easy and it is possible to re-convert to notebooks whenever it is desired.

**Comment by astrojuanlu** [comment]
...and it works on Binder :scream: 

![Screenshot from 2020-06-17 12-43-42](https://user-images.githubusercontent.com/316517/84889023-79035300-b098-11ea-8e52-5150d44ad88d.png)

(from [this demo](https://github.com/mwouts/PythonDataScienceHandbook/tree/jupytext_no_ipynb#python-data-science-handbook))

what kind of sorcery is this??

**Comment by astrojuanlu** [comment]
Full-steam with Jupytext @jorgepiloto :rocket:

**Comment by astrojuanlu** [comment]
Fixed in #978, we switched to Jupytext. This, along with #977, means that we are full steam on Markdown over `.ipynb` and reST :rocket:

**Comment by astrojuanlu** [comment]
@GorgiAstro if you notice weird things with the TOC extension, please let us know!
